microservices/recordings/src/Service/Encoder.php

```diff
@@ -25,292 +24,381 @@ class Encoder
-                    . str_replace('+', '', $callee)
-                    . '.mp3';
-
-                $recordingDto
-                    ->setCompanyId($company->getId())
-                    ->setCalldate($callDate)
-                    ->setType($type)
-                    ->setRecorder($recorder)
-                    ->setCallid($kamAccCdr->getCallid())
-                    ->setDuration(ceil($duration))
-                    ->setCaller($kamAccCdr->getCaller())
-                    ->setCallee($kamAccCdr->getCallee())
-                    ->setRecordedFileBaseName($baseName)
-                    ->setRecordedFilePath($convertMp3);
-
-
-                $isDdi = $type === RecordingInterface::TYPE_DDI && is_object($ddi);
-                $isUser = $type === RecordingInterface::TYPE_ONDEMAND && is_object($user);
-
-                if ($isDdi) {
-                    $recordingDto->setDdiId(
-                        $ddi->getId()
-                    );
-                } elseif ($isUser) {
-                    $recordingDto->setUserId(
-                        $user->getId()
-                    );
-                }
+    private function getHashId(?string $callid): string
+    {
+        return substr(md5($callid ?? ''), 0, 8);
+    }
 
-                // Store this Recording
-                $recording = $this->entityTools->persistDto($recordingDto, null, true);
-                $this->logger->info(
-                    sprintf("[Recordings][%s] Create Recordings entry with id %s\n", $hashid, (string) $recording->getId())
-                );
-                $stats['encoded']++;
-            }
+    private function storeRecording(RecordingDto $recordingDto): void
+    {
+        $hashid = $this->getHashId($recordingDto->getCallid());
+        $recording = $this->entityTools->persistDto($recordingDto, null, true);
+        $this->logger->info(
+            sprintf("[Recordings][%s] Create Recordings entry with id %s\n", $hashid, (string)$recording->getId())
+        );
+        $this->stats['encoded']++;
+    }
 
-            // Remove encoded files
-            unlink($convertWav);
+    private function getBaseName(BillableCallInterface | UsersCdrInterface $cdr, string $type, string $recorder): string
+    {
+        $callDate = $cdr->getStartTime();
+        $caller = $cdr->getCaller();
+        $callee = $cdr->getCallee();
+
+        return
+            $callDate->format('YmdHis')
+            . '_'
+            . $type
+            . '-'
+            . str_replace('+', '', $recorder)
+            . '_'
+            . str_replace('+', '', $caller)
+            . '_'
+            . str_replace('+', '', $callee)
+            . '.mp3';
+    }
+
+    /**
+     * @return UsersCdrInterface[]|BillableCallInterface[]
+     */
+    private function getAccCdrs(RawRecordingInfo $filename): array
+    {
+        $billableCalls = $this
+            ->billableCallRepository
+            ->findByCallid(
+                $filename->getCallid() ?? ''
```

Comment:
As mentioned before, this should not happen. Callid is a must in the recording and should not be nullable.